tgg/seqr_scripts/sex_check.py

```diff
@@ -0,0 +1,223 @@
+    :param str male_fstat_threshold: Fstat threshold above which a sample will be called male
+    :param str female_fstat_threshold: Fstat threshold below which a sample will be called female
+    :param str aaf_threshold: Minimum alternate allele frequency required 
+    :return: Table with imputed sex annotations
+    :rtype: Table
+    """
+    
+    logging.warn("User needs to decide on fstat thresholds for male/female by looking at fstat plots")
+
+    
+    # filter to the X chromosome and impute sex
+    mt = hl.filter_intervals(mt, [hl.parse_locus_interval(get_reference_genome(mt.locus).x_contigs[0], reference_genome=build)])   
+    sex_ht = hl.impute_sex(mt.GT, aaf_threshold=aaf_threshold, male_threshold=male_fstat_threshold, female_threshold=female_fstat_threshold)
+    mt = mt.annotate_cols(**sex_ht[mt.col_key])
+    sex_ht = mt.cols()
+    
+    # plot histogram of fstat values
+    df = sex_ht.to_pandas()
+    plt.clf()
+    plt.hist(df["f_stat"])
+    plt.xlabel("Fstat")
+    plt.ylabel("Frequency")
+    plt.axvline(male_fstat_threshold, color='blue', linestyle='dashed', linewidth=1)
+    plt.axvline(female_fstat_threshold, color='red', linestyle='dashed', linewidth=1)
+
+    outplot = outdir + "/fstat_{callset_name}.png".format(**locals())
+    with hl.hadoop_open(outplot, 'wb') as out:
+        plt.savefig(out)   
+        
+    return sex_ht  
+
+    
+def call_sex(callset: str, 
+             use_y_cov: bool=False,
+             y_cov_threshold: float=0.1,
+             male_fstat_threshold: float=0.75, 
+             female_fstat_threshold: float=0.5, 
+             aaf_threshold: float=0.05,
+             call_rate_threshold: float=0.25) -> hl.Table:
+    
+    """
+    Calls sex for the samples in a given callset and exports results file to the callset directory
+    :param str callset: String of full matrix table path for the callset
+    :param bool use_y_cov: Set to True to calculate and use chrY coverage for sex inference
+    :param float y_cov_threshold: Coverage on chrY above which supports male call
+    :param str male_fstat_threshold: Fstat threshold above which a sample will be called male
+    :param str female_fstat_threshold: Fstat threshold below which a sample will be called female
+    :param str aaf_threshold: Minimum alternate allele frequency required 
+    :param float call_rate_threshold: Minimum required call rate
+    :return: Table with sex annotations
+    :rtype: Table
+    """
+
+    # read in matrix table and define output directory
+    # will need to generalize more for gnomad methods
+    outdir = dirname(callset)
+    mt_name = callset.split("/")[-1].strip("\.mt")
+    logging.info("Reading matrix table for callset: {callset}".format(**locals()))
+    logging.info("Using chromosome Y coverage? {use_y_cov}".format(**locals()))
+
+    mt = hl.read_matrix_table(callset)
+    
+    # filter to SNVs and biallelics
+    # filter to biallelics might be a function
```

Comment:
```suggestion
    # Filter to SNVs and biallelics
```

Reply:
I found from `gnomad.utils.annotations import bi_allelic_expr`, but this is fine